### PR TITLE
Added in the Disabling Timer0 and re-enabling Timer0

### PR DESCRIPTION
**Pull request by mfeist**

### CODE REVIEW

File: StepperMotor/StepperMotor.c [hunk 1]
```diff
@@ -74,13 +74,16 @@ void StepperMotor_StepForward(void)
 	
 	if(StepperMotor_GlobalPosition == SYRINGE_LENGTH)
 	{
```
Comment (sonnle) [inline StepperMotor/StepperMotor.c:76]: If you call LEDClear either way, it should just be outside of the if loop

File: StepperMotor/StepperMotor.c [hunk 2]
```diff
@@ -161,15 +164,18 @@ void TIMER1_IRQHandler(void)
...
 			StepperMotor_CurrentBasalDose++; // Keep track of current dosing
 			StepperMotor_StepForward();
 			break;
 		case Bolus_Status:
+      BasalDose_TimingDisable();
 			StepperMotor_CurrentBolusDose++; 
 			StepperMotor_StepForward();
 			break;
 		case Backward_Status:
+      BasalDose_TimingDisable();
```
Comment (sonnle) [inline StepperMotor/StepperMotor.c]: Why do you need 3 Timing Disables? I don't think this fixes it, you should disable when you enable timer1, then enable when you disable timer0

File: StepperMotor/StepperMotor.c (hunk 1 above)
Comment (mfeist) [inline StepperMotor/StepperMotor.c:76]: LEDClear is only used for these 2 scenarios.  It shouldn't be called if the stepper motor is going to continue to spin.  I can put it back in the main loop, but then I will also move the LEDClear from the StepperMotor_StepBackward into the main as well to keep consistency.

File: StepperMotor/StepperMotor.c (hunk 2 above)
Comment (mfeist) [inline StepperMotor/StepperMotor.c]: So you're saying that the timer0 disable should be in the timer0 IRQ handler as well as the EINT3 IRQ Handler.

Comment (sonnle) [inline StepperMotor/StepperMotor.c]: No, it should be in the code section that goes after it. Like before we would disable timer0 during the first iteration through the timer1 irq handler which was the steppermotor functions.

Comment (mfeist) [inline StepperMotor/StepperMotor.c]: In the StepperMotor_SpinEnable, the first line that function is BasaDose_TimingDisable. So, it is  already disabling Timer0.  So it shouldn't be triggering multiple interrupts. But I moved the disable into the main loop in the in the switch statments for empty and full

**Pull request merged**